src/transformers/models/moshi/modeling_moshi.py

```diff
@@ -0,0 +1,2792 @@
+            have their past key value states given to this model) of shape `(batch_size, 1)` instead of all `input_ids`
+            of shape `(batch_size, sequence_length)`.
+        inputs_embeds (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`, *optional*):
+            Optionally, instead of passing `input_ids` you can choose to directly pass an embedded representation. This
+            is useful if you want more control over how to convert `input_ids` indices into associated vectors than the
+            model's internal embedding lookup matrix.
+        use_cache (`bool`, *optional*):
+            If set to `True`, `past_key_values` key value states are returned and can be used to speed up decoding (see
+            `past_key_values`).
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+        cache_position (`torch.LongTensor` of shape `(sequence_length)`, *optional*):
+            Indices depicting the position of the input sequence tokens in the sequence. Contrarily to `position_ids`,
+            this tensor is not affected by padding. It is used to update the cache in the correct position and to infer
+            the complete sequence length.
+"""
+
+
+class MoshiDepthDecoder(MoshiPreTrainedModel, GenerationMixin):
+    """
+    Transformer depth decoder consisting of *config.depth_num_hidden_layers* layers. Each layer is a [`MoshiTransformerLayer`]
+
+    Args:
+        config: MoshiConfig
+    """
+
+    def __init__(self, config: MoshiConfig):
+        original_hidden_size = config.hidden_size
+
+        config = copy.deepcopy(config)
+        for key, value in config.to_dict().items():
+            if len(key) >= 6 and "depth_" == key[:6]:
+                config.__setattr__(key[6:], value)
+        super().__init__(config)
+
+        self.text_embed_tokens = nn.Embedding(config.vocab_size + 1, config.depth_hidden_size)
+
+        # the last codebook is never used as input
+        self.embed_tokens = nn.ModuleList(
+            [
+                nn.Embedding(config.audio_vocab_size + 1, config.depth_hidden_size)
+                for _ in range(config.num_codebooks - 1)
+            ]
+        )
+
+        self.input_projections = MoshiFlexibleLinear(
+            original_hidden_size, config.depth_hidden_size, config.num_codebooks
+        )
+
+        self.layers = nn.ModuleList(
+            [
+                MoshiDecoderLayer(config, layer_idx, use_flexible_linear=True, use_rope=False)
```

Comment:
`use_flexible_linear` is always True, I don't see where it's False so not needed for the DecodeLayer class no?

Reply:
It's actually quite the opposite, `MoshiDecoderLayer` has `use_flexible_linear=False` by default (in the main decoder thus). It's used only in the depth decoder.

Reply:
yep sorry